### DISCUSSION TITLE
Visualisations

### DISCUSSION THREAD

**Discussion by ChristianLieven** [micom-dev/micom#46] [Show and tell]
Hey MICOM-munity
I've made some very basic Jupyter Notebook-friendly visualizations in Altair and D3 to help with getting a better overview of the fluxes in a community solution, specifically exchange fluxes.

Here's what I've got:

- Simple barchart showing the relative abundance:
<img width="653" alt="Screenshot 2021-06-17 at 11 15 51" src="https://user-images.githubusercontent.com/8027271/122907926-e739c000-d353-11eb-8667-e1d037699d31.png">

- Production and uptake of metabolites (read: flux through exchanges) by the community:
<img width="602" alt="Screenshot 2021-06-17 at 11 17 56" src="https://user-images.githubusercontent.com/8027271/122907921-e6a12980-d353-11eb-8034-c4789d177933.png">

- Production and uptake of metabolites (read: flux through exchanges) by individual members:
<img width="225" alt="Screenshot 2021-06-17 at 11 16 13" src="https://user-images.githubusercontent.com/8027271/122907918-e6089300-d353-11eb-8dba-e15cd659c34a.png">

- Chord-Plot that shows cross-feeding interactions:
<img width="707" alt="Screenshot 2021-06-17 at 11 18 18" src="https://user-images.githubusercontent.com/8027271/122907911-e4d76600-d353-11eb-955c-594e8a62ac10.png">

- Network-Plot that shows cross-feeding interactions:
<img width="697" alt="Screenshot 2021-06-17 at 11 19 15" src="https://user-images.githubusercontent.com/8027271/122907908-e3a63900-d353-11eb-8d03-aa4963433b0f.png">

If there is interest, I'd be happy to share my code. I'm not sure if I have the time to go through a fully-fledged PR though and may require some guidance setting up a comfortable development environment.

Cheers,

Christian

**Comment by cdiener** [comment]
I think just having some example notebooks as a Gist or repo would be great. You might even attach them here. Visualizations for active taxon-level active pathways and cross-feeding or competition are also on my roadmap. Does ALTAIR export to vega or vega-lite javascript? That could be a good strategy to design new visualizations for MICOM.

One note because I see it here: MICOM tends to predict a lot of cycling for SCFAs and I think that might be spurious to a certain degree. In the gut or any experimental setup, those usually get absorbed by the gut epithelial cells or are too volatile to be reabsorbed by microbes (especially butyrate and propionate are pretty much gases). In our experimental setup, we also observe that the pure production flux (sum a_i v^ex_i) usually agrees better with the actual flux which indicates that MICOM probably overestimates the rate of re-import for those compounds.

**Reply by ChristianLieven** [reply]
Hey @cdiener,

I'll try to bundle these nicely for you in a notebook and share them either here or as a Gist! 

ALTAIR is in one way or another using vega I believe so that shouldn't be a problem. But before everything worked nicely with Altair I thought I needed to do some tidying of the DataFrames output by Micom. I had a fairly quick and dirty approach to implementing this so there's probably a better way.

Good to know about the observations on cycling you made! I thought it likely to be a side-effect of the uncurated models I used in the example.

**Comment by Amit-H** [comment]
Hi @ChristianLieven, 

Would you still happen to have the notebook for these visualisations, and if so, would you be comfortable sharing it? These look really great!

Thank you (and no worries if not)!
